# HG changeset patch
# Parent 8bd4a45d63880328ecbce1e558439084a360c065
# User sbr
# Date 1348905329 -7200

Write cargoes abbrev in white if rating colour is really dark (unneeded with default cargoes rating colours).

diff -r 8bd4a45d6388 src/station_gui.cpp
--- a/src/station_gui.cpp	Sat Sep 29 09:43:41 2012 +0200
+++ b/src/station_gui.cpp	Sat Sep 29 12:28:58 2012 +0200
@@ -455,8 +455,9 @@
 				if (widget >= WID_STL_CARGOSTART) {
 					const CargoSpec *cs = _sorted_cargo_specs[widget - WID_STL_CARGOSTART];
 					int cg_ofst = HasBit(this->cargo_filter, cs->Index()) ? 2 : 1;
+					TextColour tc = (cs->rating_colour < 0x05 || cs->rating_colour == 0x10 || cs->rating_colour == 0x46 || cs->rating_colour == 0x68) ? TC_WHITE : TC_BLACK;
 					GfxFillRect(r.left + cg_ofst, r.top + cg_ofst, r.right - 2 + cg_ofst, r.bottom - 2 + cg_ofst, cs->rating_colour);
-					DrawString(r.left + cg_ofst, r.right + cg_ofst, r.top + cg_ofst, cs->abbrev, TC_BLACK, SA_HOR_CENTER);
+					DrawString(r.left + cg_ofst, r.right + cg_ofst, r.top + cg_ofst, cs->abbrev, tc, SA_HOR_CENTER);
 				}
 				break;
 		}
